monai/data/utils.py

```diff
@@ -242,8 +242,24 @@ def worker_init_fn(worker_id: int) -> None:
 
     """
     worker_info = torch.utils.data.get_worker_info()
-    if hasattr(worker_info.dataset, "transform") and hasattr(worker_info.dataset.transform, "set_random_state"):
-        worker_info.dataset.transform.set_random_state(worker_info.seed % (2 ** 32))
+    set_rnd(worker_info.dataset, seed=worker_info.seed)
```

Comment:
this is an enhancement to `set_random_state` for all the dataset properties, so that it supports nested dataset, such as: `Dataset(Dataset([data_1, data_2, ...], transform=random_transform_1), transform=random_transform_2)` @Nic-Ma it's useful for the patch dataset